src/compiler/corePublic.ts

```diff
@@ -114,25 +114,31 @@ namespace ts {
 
     /* @internal */
     namespace NativeCollections {
-        declare const Map: MapConstructor | undefined;
-        declare const Set: SetConstructor | undefined;
+        declare const self: any;
+
+        const globals = typeof globalThis !== "undefined" ? globalThis
+                  : typeof global !== "undefined" ? global
+                  : typeof self !== "undefined" ? self
+                  : undefined;
 
         /**
          * Returns the native Map implementation if it is available and compatible (i.e. supports iteration).
          */
         export function tryGetNativeMap(): MapConstructor | undefined {
             // Internet Explorer's Map doesn't support iteration, so don't use it.
+            const gMap = globals.Map;
             // eslint-disable-next-line no-in-operator
-            return typeof Map !== "undefined" && "entries" in Map.prototype && new Map([[0, 0]]).size === 1 ? Map : undefined;
+            return typeof gMap !== "undefined" && "entries" in gMap.prototype && new gMap([[0, 0]]).size === 1 ? gMap : undefined;
         }
 
         /**
          * Returns the native Set implementation if it is available and compatible (i.e. supports iteration).
          */
         export function tryGetNativeSet(): SetConstructor | undefined {
             // Internet Explorer's Set doesn't support iteration, so don't use it.
+            const gSet = globals.Set;

```

Comment:
`globals` is possibly `undefined`.
```suggestion
            const gSet = globals?.Set;
```